### PR TITLE
Add Server Preset guide

### PR DESCRIPTION
**Pull request by eddeee888**
## Description

This PR adds a new guide for server preset

## Type of change

- [x] Adding a guide on the GraphQL Code Generator website

## Checklist:

- [x] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project

<details>
<summary>Other unrelated checklist items</summary>
- [ ] I have performed a self-review of my own code (No code change)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
</details>

### CODE REVIEW

File: website/src/pages/docs/guides/_meta.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "svelte": "Svelte / Kit",
   "flutter-freezed": "Dart/Flutter",
   "graphql-server-apollo-yoga": "Apollo Server / GraphQL Yoga",
+  "graphql-server-apollo-yoga-with-server-preset": "Apollo Server / GraphQL Yoga with Server Preset",
```
Comment (eddeee888) [inline website/src/pages/docs/guides/_meta.json:7]: I manually added this line. Without this, the guide title does not show up correctly on the left-hand side. 

Is this the right place? I'm not sure if this file is generated or not :)

File: website/src/pages/docs/guides/graphql-server-apollo-yoga-with-server-preset.mdx [hunk 2]
```diff
@@ -0,0 +1,349 @@
+import { PackageCmd, Callout, Tabs, Tab } from '@theguild/components'
+
+# Guide: GraphQL Yoga / Apollo Server with Server Preset
```
Comment (eddeee888) [inline website/src/pages/docs/guides/graphql-server-apollo-yoga-with-server-preset.mdx:3]: This is a separate guide from https://the-guild.dev/graphql/codegen/docs/guides/graphql-server-apollo-yoga. I want to keep them separate for now, and put a callout in the original article to suggest this article instead?

Alternatively, we could update the original title but does abruptly change the content cause confusion and (maybe) SEO issues? Happy to hear thoughts/advices here.

File: website/src/pages/docs/guides/graphql-server-apollo-yoga-with-server-preset.mdx [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+│   │   │   ├── schema.graphql
+│   │   ├── resolvers.generated.ts         # Entirely generated by codegen
+│   │   ├── typesDefs.generated.ts         # Entirely generated by codegen
+│   │   ├── types.generated.ts             # Entirely generated by codegen
+```
+
+#### Generated Files Overview
+
+- `types.generated.ts`: TypeScript types generated by `@graphql-codegen/typescript` and `@graphql-codegen/typescript-resolvers`
+- `typeDefs.generated.ts`: Static TypeScript Schema AST to be used by the server
```
Comment (eddeee888) [inline website/src/pages/docs/guides/graphql-server-apollo-yoga-with-server-preset.mdx:127]: This feature is not relased yet: https://github.com/eddeee888/graphql-code-generator-plugins/pull/51

I'll do some final testings today

Comment (eddeee888) [reply website/src/pages/docs/guides/graphql-server-apollo-yoga-with-server-preset.mdx:127]: This is done

File: website/src/pages/docs/guides/_meta.json (hunk 1 above)
Comment (n1ru4l) [reply website/src/pages/docs/guides/_meta.json:7]: yes it is fine, we are using nextra 😇 see https://nextra.site/docs/guide/organize-files#_metajson

File: website/src/pages/docs/guides/graphql-server-apollo-yoga-with-server-preset.mdx (hunk 2 above)
Comment (n1ru4l) [reply website/src/pages/docs/guides/graphql-server-apollo-yoga-with-server-preset.mdx:3]: I agree with keeping both in the meantime until we have some more feedback/usage of the server preset. Lets also add a Callout to the other doc page for encouraging people to try this approach!

**Review by n1ru4l — APPROVED** [review]

File: website/src/pages/docs/guides/graphql-server-apollo-yoga-with-server-preset.mdx (hunk 2 above)
Comment (eddeee888) [reply website/src/pages/docs/guides/graphql-server-apollo-yoga-with-server-preset.mdx:3]: This is done in https://github.com/dotansimha/graphql-code-generator/pull/8858/commits/bda2c1d56d95aa82b35fd64f0baf766b0205cb52 and looks something like this:

Screen Shot 2023-02-01 at 10 59 39 pm

Happy to make changes to the text or position if needed :)

**Pull request merged**